mockld/events_service.go

```diff
@@ -64,6 +73,14 @@ func (s *EventsService) postEvents(w http.ResponseWriter, r *http.Request) {
 		s.logger.Printf("unable to read request body")
 		return
 	}
+
+	s.lock.Lock()
+	hostTime := s.hostTimeOverride
+	s.lock.Unlock()
+	if !hostTime.IsZero() {
+		w.Header().Set("Date", hostTime.UTC().Format(http.TimeFormat))
```

Comment:
This is the thing we can't simulate with integration-harness. And in some SDKs, we can't simulate it in unit tests either, at least not when we're doing actual HTTP in the tests— some embedded HTTP server frameworks don't allow the `Date` header to be overridden. So in those SDKs, we've only been able to test it by completely mocking out the HTTP client piece, which isn't ideal.